metal/cache.h

```diff
@@ -11,51 +11,26 @@
  */
 #include <stdint.h>
 
-struct metal_cache;
-
-struct __metal_cache_vtable {
-    void (*init)(struct metal_cache *cache, int ways);
-    int (*get_enabled_ways)(struct metal_cache *cache);
-    int (*set_enabled_ways)(struct metal_cache *cache, int ways);
-};
-
-/*!
- * @brief a handle for a cache
- */
-struct metal_cache {
-    const struct __metal_cache_vtable *vtable;
-};
-
 /*!
- * @brief Initialize a cache
- * @param cache The handle for the cache to initialize
- * @param ways The number of ways to enable
- *
- * Initializes a cache with the requested number of ways enabled.
+ * @brief Initialize L2 cache controller
+ * @param None
+ * @return 0 If no error
  */
-__inline__ void metal_cache_init(struct metal_cache *cache, int ways) {
```

Comment:
We should not delete this API, as part of a release update, which is what kiwi is.
Keep the API and its signature, but the vtable and structure can be remove.
It should then call the new APIs, in this case metal_l2cache_get_enabled_ways(), metal_l2cache_set_enabled_ways()
Also add a comment that this API will be deprecated in the next release.